packages/frontend/src/components/create-campaign/form/kpi-step/goal-inputs/index.tsx

```diff
@@ -0,0 +1,131 @@
+import type { CampaignPayload } from "@/src/types";
+import {
+    NumberInput,
+    SliderInput,
+    type NumberFormatValues,
+} from "@metrom-xyz/ui";
+import { useState, type ChangeEvent } from "react";
+import { useTranslations } from "next-intl";
+import { useDebounce } from "react-use";
+
+import styles from "./styles.module.css";
+
+interface GoalInputsProps {
+    error?: boolean;
+    kpiSpecification?: CampaignPayload["kpiSpecification"];
+    onLowerUsdTargetChange: (value: number | undefined) => void;
+    onUpperUsdTargetChange: (value: number | undefined) => void;
+    onMinimumPayoutPercentageChange: (value: number) => void;
+}
+
+export interface NumberInputValues {
+    raw?: NumberFormatValues["floatValue"];
+    formatted?: NumberFormatValues["formattedValue"];
+}
+
+export function GoalInputs({
+    error,
+    kpiSpecification,
+    onLowerUsdTargetChange,
+    onUpperUsdTargetChange,
+    onMinimumPayoutPercentageChange,
+}: GoalInputsProps) {
+    const t = useTranslations("newCampaign.form.kpi");
+
+    const [minimumPayoutPercentage, setMinimumPayoutPercentage] = useState(
+        kpiSpecification?.minimumPayoutPercentage || 0,
+    );
+    const [lowerUsdTargetRaw, setLowerUsdTargetRaw] = useState<
+        NumberInputValues | undefined
+    >(() => {
+        if (kpiSpecification) {
+            return {
+                raw: kpiSpecification.goal.lowerUsdTarget,
+                formatted: kpiSpecification.goal.lowerUsdTarget.toString(),
+            };
+        }
+        return undefined;
+    });
+    const [upperUsdTargetRaw, setUpperUsdTargetRaw] = useState<
+        NumberInputValues | undefined
+    >(() => {
+        if (kpiSpecification) {
+            return {
+                raw: kpiSpecification.goal.upperUsdTarget,
+                formatted: kpiSpecification.goal.upperUsdTarget.toString(),
+            };
+        }
+        return undefined;
+    });
+
+    useDebounce(
+        () => {
+            onLowerUsdTargetChange(lowerUsdTargetRaw?.raw);
+        },
+        500,
+        [lowerUsdTargetRaw],
+    );
+    useDebounce(
```

Comment:
Can we add a new line here?

Reply:
Addressed in 882edcfff8120a77c162cce2b4880b7c4ce5c1bf.